parking/backend/engine/alloc_engine.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import Optional
+from geopy.distance import distance
+from parking.backend.db.dbaccess import DbAccess
+from parking.shared.rest_models import ParkingLot
+from parking.shared.location import Location
+from parking.shared.ws_models import ParkingRequestMessage
+
+logger = logging.getLogger('backend')
+
+
+class AllocationEngine():
+    def __init__(self, dba: DbAccess, user_sessions: 'UserSessions') -> None:
+        self.dba = dba
+        self.user_sessions = user_sessions
+
+    async def handle_allocation_request(self, user_id: str, request: ParkingRequestMessage) -> Optional[ParkingLot]:
+        rejections = self.user_sessions.get_user(user_id).rejections
```

Comment:
rejections is a liiiiiittle bit vague. I'm guessing it's parking lots that the user has rejected, rather than rejections of the user by the engine?

Reply:
Yes, could add a comment or rename to user_rejections.

Reply:
I'll rename to user_rejections